(venv) PS C:\Users\HP\ecommerce_project> pytest
============================================== test session starts ==============================================
platform win32 -- Python 3.11.9, pytest-8.4.2, pluggy-1.6.0
rootdir: C:\Users\HP\ecommerce_project
plugins: django-4.11.1
collected 0 items / 2 errors

==================================================== ERRORS ===================================================== 
_________________________________ ERROR collecting catalog/tests/test_models.py _________________________________ 
catalog\tests\test_models.py:2: in <module>
    from django.contrib.auth.models import User
venv\Lib\site-packages\django\contrib\auth\models.py:5: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
venv\Lib\site-packages\django\contrib\auth\base_user.py:43: in <module>
    class AbstractBaseUser(models.Model):
venv\Lib\site-packages\django\db\models\base.py:131: in __new__
    app_config = apps.get_containing_app_config(module)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
venv\Lib\site-packages\django\apps\registry.py:260: in get_containing_app_config
    self.check_apps_ready()
venv\Lib\site-packages\django\apps\registry.py:137: in check_apps_ready
    settings.INSTALLED_APPS
venv\Lib\site-packages\django\conf\__init__.py:81: in __getattr__
    self._setup(name)
venv\Lib\site-packages\django\conf\__init__.py:61: in _setup
    raise ImproperlyConfigured(
E   django.core.exceptions.ImproperlyConfigured: Requested setting INSTALLED_APPS, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
_________________________________ ERROR collecting catalog/tests/test_views.py __________________________________ 
catalog\tests\test_views.py:1: in <module>
    from rest_framework.test import APIClient
venv\Lib\site-packages\rest_framework\test.py:138: in <module>
    class APIRequestFactory(DjangoRequestFactory):
venv\Lib\site-packages\rest_framework\test.py:139: in APIRequestFactory
    renderer_classes_list = api_settings.TEST_REQUEST_RENDERER_CLASSES
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
venv\Lib\site-packages\rest_framework\settings.py:220: in __getattr__
    val = self.user_settings[attr]
          ^^^^^^^^^^^^^^^^^^
venv\Lib\site-packages\rest_framework\settings.py:211: in user_settings
    self._user_settings = getattr(settings, 'REST_FRAMEWORK', {})
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
venv\Lib\site-packages\django\conf\__init__.py:81: in __getattr__
    self._setup(name)
venv\Lib\site-packages\django\conf\__init__.py:61: in _setup
    raise ImproperlyConfigured(
E   django.core.exceptions.ImproperlyConfigured: Requested setting REST_FRAMEWORK, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
============================================ short test summary info ============================================ 
ERROR catalog/tests/test_models.py - django.core.exceptions.ImproperlyConfigured: Requested setting INSTALLED_APPS, but settings are not configure...
ERROR catalog/tests/test_views.py - django.core.exceptions.ImproperlyConfigured: Requested setting REST_FRAMEWORK, but settings are not configure...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! 
=============================================== 2 errors in 0.70s =============================================== 
(venv) PS C:\Users\HP\ecommerce_project> 